### PR TITLE
feat: handle redirects in Next middleware

### LINKED ISSUE #9221: We can create redirects for both product and SDK docs (preferably with regexes) that don't count against our Vercel routes limit

### PR DESCRIPTION
**Pull request by a-hariti**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

Handle redirects in custom middleware to avoid Vercel redirect limits

Closes #9221

## Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

## Extra resources

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -25,3106 +25,8 @@ const nextConfig = {
   },
 
   redirects() {
```
Comment (lforst) [inline next.config.js:27]: I think we can remove the redirects function completely

**Comment by lforst** [comment]
Our bill may explode but let's see.

File: next.config.js (hunk 1 above)
Comment (a-hariti) [reply next.config.js:27]: Yes, we can.

I kept it there to signal the intention of having no redirects in the config

**Comment by a-hariti** [comment]
> Our bill may explode but let's see.

I hope not 😆 , this is supposed to have the opposite effect

File: next.config.js (hunk 1 above)
Comment (lforst) [reply next.config.js:27]: I'd rather remove it. Keeping it around seems unnecessary. People are still allowed to define static redirects.

**Review by lforst — APPROVED** [review]

**Comment by lforst** [comment]
Awesome. Thank you!

**Comment by mitsuhiko** [comment]
Can confirm. Bill did explode.

**Comment by a-hariti** [comment]
Is the `/changelog/` path getting more traffic or is the implementation buggy?

Also I see no redirects for `/changelog`, so we can safely exclude from the matcher. But it would be helpful to know why it happened first

**Pull request merged**